contrib/kythe/src/python/pants/contrib/kythe/tasks/indexable_java_targets.py

```diff
@@ -6,25 +6,36 @@
                         unicode_literals, with_statement)
 
 from pants.backend.jvm.targets.jvm_target import JvmTarget
-from pants.build_graph.target_scopes import Scopes
+from pants.build_graph.target_scopes import Scope
+from pants.subsystem.subsystem import Subsystem
 
 
-class IndexableJavaTargets(object):
+class IndexableJavaTargets(Subsystem):
   """Determines which java targets Kythe should act on."""
+  options_scope = 'kythe-java-targets'
 
   @classmethod
-  def get(cls, context):
+  def register_options(cls, register):
+    super(IndexableJavaTargets, cls).register_options(register)
+    register('--exclude-scopes', type=list, member_type=str, fingerprint=True,
+             help='Dependency scopes to exclude from indexing.')
+    register('--recursive', type=bool, fingerprint=True,
```

Comment:
Related to #4973. Would it be a good idea to split out a separate (scoped) Subsystem to control transitivity?

Reply:
Oh yeah! How about I do that in a followup, since no one but me is using this plugin, and I want to move forward.

Reply:
There might be unintended effects of this, and there will certainly be some design debate. So I'd rather do it separately.